tests/src/rules/no-cycle.js

```diff
@@ -88,6 +93,16 @@ ruleTester.run('no-cycle', rule, {
       code: 'import { bar } from "./depth-three-indirect"',
       errors: [error(`Dependency cycle via ./depth-two:1=>./depth-one:1`)],
     }),
+    // test({
```

Comment:
i assume these should be uncommented

Reply:
It should. Errors for `import()` with `no-cycle` aren't working because of [the workaround for `flow` import resolution in `eslint@<5.x`](https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/no-cycle.js#L37).

Reply:
That seems like something worth fixing. Why would a flow workaround impact `import()`, which is actual JS?

Reply:
That seems like something worth fixing. Why would a flow workaround impact `import()`, which is actual JS?

Reply:
That seems like something worth fixing. Why would a flow workaround impact `import()`, which is actual JS?

Reply:
That seems like something worth fixing. Why would a flow workaround impact `import()`, which is actual JS?

Reply:
Fixed by using `importer.importKind` instead of `sourceNode.parent && sourceNode.parent.importKind`. Would that break anything? I added a few more tests for `no-cycle` that import multiple variables together just to be careful.

Reply:
Worth noting - `no-cycle` was also not working for static `import` with `babel-eslint` because of this, which is why I added a test case for static `import` with the `babel-eslint` parser.